### PR TITLE
Implement `tapOnNotificationByIndex` and `tapOnNotificationBySelector` on iOS

### LINKED ISSUE #319: Tap on notification by text and by index on iOS

### PR DESCRIPTION
**Pull request by bartekpacia**
This PR:
- fixes #319
- adds new `closeHeadsUpNotification()` method (iOS only)
- refactor of directory structure in the example app

### CODE REVIEW

File: AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift [hunk 1]
```diff
@@ -70,6 +52,44 @@ class PatrolAutomation {
       start.press(forDuration: 0.1, thenDragTo: end)
     }
   }
+  
+  // MARK: General UI interaction
```
Comment (shilangyu) [inline AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift:56]: MARK?

File: AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift [hunk 2]
```diff
@@ -178,65 +196,82 @@ class PatrolAutomation {
...
   }
   
+  // MARK: Notifications
+  
+  func openNotifications() {
+    // TODO: Check if works on iPhones without notch
+    
+    runAction("opening notifications") {
+      let start = self.springboard.coordinate(withNormalizedOffset: CGVector(dx: 0.5, dy: 0.01))
+      let end = self.springboard.coordinate(withNormalizedOffset: CGVector(dx: 0.5, dy: 0.6))
```
Comment (shilangyu) [inline AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift:206]: Oh boy, I dont know iOS, but isnt this gesture iOS-version-related? Is there really no method for doing it?

File: AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift [hunk 3]
```diff
@@ -178,65 +196,82 @@ class PatrolAutomation {
...
-      app.textFields[text].firstMatch.typeText(data)
+  
+  func tapOnNotification(by index: Int) {
+    runAction("tapping on notification at index \(index)") {
+      let cells = self.springboard.buttons.matching(identifier: "NotificationCell").allElementsBoundByIndex
+      if !cells.indices.contains(index) {
+        Logger.shared.e("notification at index \(index) doesn't exist")
+      }
+      
+      cells[index].tap()
```
Comment (shilangyu) [inline AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift:244]: Do you still want to tap if it does not exist?

**Review by shilangyu — APPROVED** [review]

File: AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift (hunk 3 above)
Comment (bartekpacia) [reply AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift:244]: Ah, a `return` should be there.

File: AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift (hunk 1 above)
Comment (bartekpacia) [reply AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift:56]: Yeah, it's quite nice:
Screenshot 2022-10-01 at 5 30 26 PM
Screenshot 2022-10-01 at 5 30 41 PM

File: AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift (hunk 2 above)
Comment (bartekpacia) [reply AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift:206]: It is, but this behavior seems to be consistent across the newest iOS versions.

> Is there really no method for doing it?

Unfortunately no. That's all we have at our disposal.

File: AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift [hunk 4]
```diff
@@ -178,65 +196,83 @@ class PatrolAutomation {
...
-
-  func enterText(_ data: String, by text: String, inApp appId: String) {
-    runAction("entering text \"\(text)\"") {
-      let app = XCUIApplication(bundleIdentifier: appId)
-      app.textFields[text].firstMatch.typeText(data)
+  
+  func tapOnNotification(by index: Int) {
+    runAction("tapping on notification at index \(index)") {
+      let cells = self.springboard.buttons.matching(identifier: "NotificationCell").allElementsBoundByIndex
+      if !cells.indices.contains(index) {
```
Comment (shilangyu) [inline AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift:240]: Use `guard...else` since its more idiomatic in swift

**Review by shilangyu — APPROVED** [review]

File: AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift (hunk 4 above)
Comment (bartekpacia) [reply AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift:240]: right, thanks

Comment (bartekpacia) [reply AutomatorServer/ios/AutomatorServerUITests/PatrolAutomation.swift]: Swift 🧡

**Pull request merged**